3.3.0 blog post announcement (#105)
sourcegraph/about: blogposts/announcing-sourcegraph-3.3.md

---
title: 'Sourcegraph 3.3: Improving the user experience for site admins'
author: Ryan Blunden
publishDate: 2019-04-22T10:00-07:00
tags: [
  blog
]
slug: sourcegraph-3
heroImage: /sourcegraph-mark.png
published: true
---

*Sourcegraph is an open-source, self-hosted, cross-repository code search and navigation tool, with an efficient web interface and "feels-like-native" integration into your code host. [Install or upgrade Sourcegraph](#install-or-upgrade).*

---

<h3 style="text-align: center">Deploy or upgrade Sourcegraph 3.3</h3>

<div style="text-align: center">

  [Local](https://docs.sourcegraph.com/#quickstart-guide) | [AWS](https://github.com/sourcegraph/deploy-sourcegraph-aws) | [DigitalOcean](https://marketplace.digitalocean.com/apps/sourcegraph?action=deploy&refcode=48dfb3ccb51c) | [Kubernetes](https://github.com/sourcegraph/deploy-sourcegraph)

Want to try pre-release development builds? Change the Docker image tag to  `sourcegraph:insiders`.

</div>

---

This release focusses on making Sourcegraph easier to set up, configure and manage repositories.

> We’ve completely overhauled how repositories are selected and kept in-sync, benefiting everyone, but especially organizations with thousands (or tens of thousands) of repositories. We’ve also added Sentry to our list of third party developer tool integrations and improved Swift code intelligence and navigation.

Reviewer: ```suggestion
We’ve completely overhauled how repositories are mirrored from your code host so that it is easier to configure and more reliable. This benefits everyone, but especially organizations with thousands (or tens of thousands) of repositories. We’ve also added Sentry to our list of third party developer tool integrations and improved Swift code intelligence and navigation.
```

Read on for the details, and thanks to our customers and community for reporting issues and providing feedback. You’re helping to make each Sourcegraph release the best one yet!

**Release highlights**:

[**🛠️ New! Config based repository selection**](#repo-config)<br />
Repositories now chosen via explicit lists and query rules in the external service config.

**[🔌 New Sentry integration](#sentry) and [Bitbucket Server fixes](#bitbucket-fixes)**<br />
Jump to Sentry from Sourcegraph and other integrations improvements.

[**🧭 More precise code intelligence**](#code-intel)<br />
Symbols scoped to the current file and improved Swift support.

[**🚢  Deployments secure by default with TLS pre-configured**](#deployments)<br />
TLS on all Terraform plans and improved AWS deployment docs.

> [**Making Sourcegraph a part of your dev tools stack**](#grok-sourcegraph)<br />📣

Reviewer: ```suggestion
[**📣 Making Sourcegraph a part of your dev tools stack**](#grok-sourcegraph)<br />
```

Tips for integrating Sourcegraph and code search into your team for better code.

[**📖 New docs design**](#docs)<br />
Better UI, new page navigation, and docs improvements for site admins.

> Also in this release:

Reviewer: Doesn't seem necessary
```suggestion
```

[**📝 Changelog**]()<br />
Every detail that changed in this release

[**🎖️ Thank you**]()<br />
Sourcegraph couldn’t be what it is without the community

<div id="repo-config"></div>

## New repository configuration for GitHub, GitLab, and Bitbucket Server

Prior to 3.3, integrating with a code host and selecting which repositories to use was a manual process:

> ![](/blog/old-repository-sync.png)

Reviewer: broken image

Author: It's not broken when rendered by Gatsby.

While it was simple, it presented some problems:
Sourcegraph would fetch and list all repositories accessible to the token which often included both an account’s private repositories, as well as the repositories from associated GitHub orgs.

Once dealing with more than a few dozen repositories (most of our customers have many thousands), manually enabling each repository was time consuming, and often resulted in scripted GraphQL queries with selection logic.

From 3.3 onwards, selecting which repositories to search and mirror from GitHub, GitLab or Bitbucket Server is now part of the external service configuration for each code host.

This configuration exclusively determines which repositories are used, eliminating the manual process of enabling, disabling and deleting individual repositories.

Additionally, if a repository is deleted from the code host or if it is no longer accessible to the provided token, it will be deleted from Sourcegraph automatically. Renames of upstream repositories are also gracefully detected and handled. 

> ![](/blog/new-repository-sync.png)

Reviewer: broken image

<div id="sentry"></div>

## New Sentry integration with the Sentry Sourcegraph extension

TODO: Replace screencast with Sourcegraph intro screen

<p class="container">
  <div style="padding:56.25% 0 0 0;position:relative;">
    <iframe src="https://player.vimeo.com/video/330644055?color=0CB6F4&amp;title=0&amp;byline=" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
  </div>
  <p style="text-align: center"><a href="https://vimeo.com/330644055" target="_blank">View on Vimeo</a></p>
</p>

[Sentry](https://sentry.io/) is an [open-source](https://github.com/getsentry/sentry) error tracking tool which captures and sends error and exception notifications instantly.. When browsing code on Sourcegraph or your code host, wouldn’t it be great if you could jump straight to the Sentry records page whenever you see error handling code, without having to manually open Sentry and find it?

> The new [Sentry Sourcegraph extension](https://sourcegraph.com/extensions/sourcegraph/sourcegraph-sentry) does just that and runs anywhere that Sourcegraph does, such as code views on GitHub/GitLab with the [Sourcegraph browser extension\(https://docs.sourcegraph.com/integration/browser_extension).

Reviewer: ```suggestion
The new [Sentry Sourcegraph extension](https://sourcegraph.com/extensions/sourcegraph/sourcegraph-sentry) does just that and runs anywhere that Sourcegraph does, such as code views on GitHub/GitLab with the [Sourcegraph browser extension](https://docs.sourcegraph.com/integration/browser_extension).
```

The initial version of the extension provides support for the most popular languages:

- Go
- Python
- Java
- JavaScript
- TypeScript

Additional languages will be added as part of each new release and if you want a particular language added, create an issue on the `sourcegraph-sentry GitHub repository`.

<div id="bitbucket-fixes"></div>

## Bitbucket Server integration and browser extension fixes

The refactoring and fixing work for integrations which started in 3.2 continues in this release, with many Bitbucket Server integration and browser extension fixes:

> - [**Bitbucket server fixes**](https://github.com/sourcegraph/sourcegraph/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed+label%3Abitbucket+milestone%3A3.3+)<br />

Reviewer: This bullet has a br after the title, but the next one doesn't. Should be consistent one way or the other.

Author: Thanks, that's been fixed
